js/ui/paygUnlockDialog.js

```diff
@@ -152,7 +146,15 @@ var PaygUnlockCodeEntry = new Lang.Class({
     },
 
     addCharacter: function(character) {
-        if (!this._enabled || !GLib.unichar_isdigit(character))
+        if (!this._enabled || !GLib.unichar_isprint(character))
+            return;
+
+        let pos = this.clutter_text.get_cursor_position();
+        let before = pos === -1 ? this._code : this._code.slice(0, pos);
+        let after = pos === -1 ? "" : this._code.slice(pos);
+        let newCode = before + character + after;
+
+        if (!Main.paygManager.validateCode(newCode, true /* partial */))
```

Comment:
Nitpick: use `//`, move the comment above the `if ()` line

Reply:
I don't think the comment is clear unless it's inline:

```js
// partial
if (!Main.paygManager.validateCode(newCode, true))
```

so I'll just remove it.